src/components/CartItem.jsx

```diff
@@ -1,30 +1,28 @@
 import React from 'react'
 import { useAuth } from '../contexts';
+import { handleIncrementDecrement, removeFromCart } from '../utils';
 
 const CartItem = ({product}) => {
 
-  const {title,imageUrl, price, quantity, count} = product;
-  const {userDetails, setUserDetails} = useAuth();
-  const {cartList} = userDetails;
+  const {title,imageUrl, price, quantity, qty} = product;
+  const {userDetails, dispatchUser} = useAuth();
+  const {isLoggedIn, cartList} = userDetails;
 
   const deleteFromCart = (product) => {
-      const temp = cartList.filter((item)=>item._id!==product._id);
-      setUserDetails({...userDetails, cartList: temp});
+      removeFromCart(product, isLoggedIn, dispatchUser);
   }  
 
   const removeCartItem=(product)=>{
-        if(product.count < 2){
-            deleteFromCart(product)
+        if(product?.qty < 2){
+            removeFromCart(product, isLoggedIn, dispatchUser);
         }
         else{
-            const temp = cartList.map((item)=>item._id==product._id?{...item, count: item.count-1}:item)
-            setUserDetails({...userDetails,cartList: temp});
+            handleIncrementDecrement(product, isLoggedIn, dispatchUser, "decrement")
```

Comment:
this function can be named as `updateCartQuantity` instead of `removeCartItem` since it's handling the increment & decrement of item quantity

Reply:
remove is only removing items from cart. increment decrement is taking care of qty count